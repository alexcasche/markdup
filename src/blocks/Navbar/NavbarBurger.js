import styled from 'styled-components'
import { applyStyleModifiers } from 'styled-components-modifiers'
import { tint } from 'polished'

const MODIFIERS = {
  open: ({ theme }) => `
    background-color: transparent;
    &:before {
      transform: rotate(-45deg) translate(-4px, 4px);
    }
    &:after {
      transform: rotate(45deg) translate(-4px, -4px);
    }
  `,
  hover: ({ theme }) => `
    background-color: ${tint(0.25, theme.colors.grey)};
    &:before, &:after {
      background-color: ${tint(0.25, theme.colors.grey)};
    }
  `,
}

const NavbarBurger = styled.span`
  position: relative;
  background-color: ${props => props.theme.colors.grey};
  width: 20px;
  height: 2px;
  transition: inherit;
  &:before,
  &:after {
    position: absolute;
    left: 0;
    top: -6px;
    content: '';
    width: inherit;
    height: inherit;
    background-color: ${props => props.theme.colors.grey};
    width: 20px;
    height: 2px;
    transition: inherit;
  }
  &:after {
    top: 6px;
  }
  ${applyStyleModifiers(MODIFIERS)};
`

export default NavbarBurger
