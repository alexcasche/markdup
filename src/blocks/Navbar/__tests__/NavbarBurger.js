import NavbarBurger from '../NavbarBurger'

describe('NavbarBurger Element', () => {
  it('renders correctly', () => {
    const tree = render(<NavbarBurger />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('adds open modifier', () => {
    const tree = render(<NavbarBurger modifiers={['open']} />).toJSON()
    expect(tree).toMatchSnapshot()
    expect(tree).toHaveStyleRule('background-color', 'transparent')
  })
})
