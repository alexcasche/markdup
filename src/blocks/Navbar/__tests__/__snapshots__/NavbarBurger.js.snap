// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavbarBurger Element adds open modifier 1`] = `
.c0 {
  position: relative;
  background-color: #ABADB3;
  width: 20px;
  height: 2px;
  -webkit-transition: inherit;
  transition: inherit;
  background-color: transparent;
}

.c0:before,
.c0:after {
  position: absolute;
  left: 0;
  top: -6px;
  content: '';
  width: inherit;
  height: inherit;
  background-color: #ABADB3;
  width: 20px;
  height: 2px;
  -webkit-transition: inherit;
  transition: inherit;
}

.c0:after {
  top: 6px;
}

.c0:before {
  -webkit-transform: rotate(-45deg) translate(-4px,4px);
  -ms-transform: rotate(-45deg) translate(-4px,4px);
  transform: rotate(-45deg) translate(-4px,4px);
}

.c0:after {
  -webkit-transform: rotate(45deg) translate(-4px,-4px);
  -ms-transform: rotate(45deg) translate(-4px,-4px);
  transform: rotate(45deg) translate(-4px,-4px);
}

<span
  className="c0"
/>
`;

exports[`NavbarBurger Element renders correctly 1`] = `
.c0 {
  position: relative;
  background-color: #ABADB3;
  width: 20px;
  height: 2px;
  -webkit-transition: inherit;
  transition: inherit;
}

.c0:before,
.c0:after {
  position: absolute;
  left: 0;
  top: -6px;
  content: '';
  width: inherit;
  height: inherit;
  background-color: #ABADB3;
  width: 20px;
  height: 2px;
  -webkit-transition: inherit;
  transition: inherit;
}

.c0:after {
  top: 6px;
}

<span
  className="c0"
/>
`;
